xwiki-rendering-syntaxes/xwiki-rendering-syntax-wikimodel/src/main/java/org/xwiki/rendering/internal/parser/wikimodel/AbstractWikiModelParser.java

```diff
@@ -91,7 +91,11 @@ protected StreamParser getLinkLabelParser()
     public XDOM parse(Reader source) throws ParseException
```

Comment:
I don't think that's the right location. Better make the already existing protection in `parse(Reader source, Listener listener, IdGenerator idGenerator)` cover more cases.